workspaces/adventure-pack/goodies/kotlin/src/traverse_level_order/Main.kt

```diff
@@ -0,0 +1,26 @@
+package traverse_level_order
+
+import common.TreeNode
+
+public fun TreeNode?.traverseLevelOrder(): Sequence<TreeNode> {
```

Comment:
I like grafting this onto the TreeNode class! That will make it really convenient to use. Also glad you went with a sequence, I definitely think sequences should be our go-to for the Kotlin goodies.

One thing to note for level order, I think when people traverse a tree in this way it's quite common to want to operate on the tree a level at a time. So rather than returning a `Sequence<TreeNode>` I think it would be more convenient to give folks a `Sequence<List<TreeNode>>` or a `Sequence<Sequence<TreeNode>>` (not sure which of those would be better so I'd be curious what you think), and then whoever wants to flatten it can do so at the callsite.